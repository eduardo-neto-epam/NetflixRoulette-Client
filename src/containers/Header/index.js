import React from 'react'
import styles from './Header.module.scss'
import classnames from 'classnames/bind'
import Logo from '../../components/Logo'
import Button from '../../components/Button'

let cx = classnames.bind(styles)

const Header = () => {
    return (
        <div className={cx('container')}>
            <Logo />
            <Button size='lg' style='glass'>+ ADD MOVIE</Button>
        </div>
    )
}

export default Header;