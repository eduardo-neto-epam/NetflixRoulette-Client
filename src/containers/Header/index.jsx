import React, { useState } from 'react';
import classnames from 'classnames/bind';
import styles from './Header.module.scss';
import Logo from '../../components/Logo/index.jsx';
import Button from '../../components/Button/index.jsx';
import Modal from '../../components/Modal/index.jsx';
import MovieForm from '../../forms/MovieForm/index.jsx';
import ErrorBoundary from '../../components/ErrorBoundary/index.jsx';

const cx = classnames.bind(styles);

const Header = () => {
  const [isModalOpen, setIsModalOpen] = useState(false);

  function handleClick() {
    setIsModalOpen(true);
  }

  function closeModal() {
    setIsModalOpen(false);
  }

  return (
    <div className={cx('container')}>
      <Logo />
      <Button size="lg" btnStyle="glass" type="button" handleClick={handleClick}>+ ADD MOVIE</Button>
      {isModalOpen && (
        <Modal onClose={closeModal}>
          <ErrorBoundary>
            <MovieForm />
          </ErrorBoundary>
        </Modal>
      )}
    </div>
  );
};

export default Header;
