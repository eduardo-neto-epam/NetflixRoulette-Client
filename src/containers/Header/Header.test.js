import renderer from 'react-test-renderer';
import React from 'react'
import Header from './index'

test('Header renders correctly', () => {
    const tree = renderer
        .create(<Header />)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });