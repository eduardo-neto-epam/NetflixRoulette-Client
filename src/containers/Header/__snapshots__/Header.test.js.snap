// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <span
      className="brand brand--bold"
    >
      netflix
    </span>
    <span
      className="brand"
    >
      roulette
    </span>
  </div>
  <button
    className="btn btn--lg btn--glass"
    onClick={[Function]}
    type="button"
  >
    + ADD MOVIE
  </button>
</div>
`;
